displayio/tilegrid.py

```diff
@@ -99,14 +99,18 @@ def __init__(
         self._flip_y = False
         self._top_left_x = 0
         self._top_left_y = 0
-        if tile_width is None:
+        if tile_width is None or tile_width == 0:
             tile_width = bitmap_width
-        if tile_height is None:
+        if tile_height is None or tile_width == 0:
             tile_height = bitmap_height
-        if tile_width > 0 and bitmap_width % tile_width != 0:
+        if tile_width < 1:
+            tile_width = 1
```

Comment:
Let's raise an exception instead. 0 makes no sense and should therefore be invalid.

Reply:
We should probably fix it in display_text first.